_source/_posts/2022-06-19-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,778 @@
+```
+
+And tried it again:
+
+```bash
+eksctl create cluster --name okta-k8s \
+--region us-west-2 \
+--zones=us-west-2a,us-west-2b,us-west-2c \
+--nodegroup-name okta-k8s-nodes \
+--node-type t2.xlarge \
+--nodes 2
+```
+
+You may have to rename the cluster. I found that sometimes even through the cluster resources have been deleted, the name is still unavailable.
+
+Perhaps you are wondering, if you got this error, why I didn't include the `--zones` flag in the original command. Was it just for the teachable moment? Not entirely. I wasn't sure if the availability in the zones is consistent enough to warrant including the specific zones in the original command.
+
+## Deploy the JHipster Kubernetes microservice to EKS
+
+Next you'll use `kubectl` to deploy the microservice to AWS EKS. Before you do that, if you haven't already, you should install a tool for monitoring the services. Both [k9s](https://github.com/derailed/k9s) and [Kubernetes Lens](https://k8slens.dev/) are great. Installing one of these tools isn't strictly necessary. You can do everything from the `kubectl` if you want, but life is short. Live a little. Use a GUI instead of the CLI. Step away from the `vim` and try an IDE. Nobody will know. You'll still be cool.
+
+Open a Bash shell and navigate to the `kubernetes` subdirectory of the project.
+
+Run the following command to update the AWS Kubernetes configuration.
+
+```bash
+aws eks update-kubeconfig --name okta-k8s
+```
+
+If you don't do this, you may see the following error.
+
+```bash
+Kubeconfig user entry is using deprecated API version client.authentication.k8s.io/v1alpha1. Run 'aws eks update-kubeconfig' to update.
+```
+
+Now you're ready to deploy the microservice. 
+
+Run the following command (again, from the `kubernetes` subdirectory).
+
+```bash
+./kubectl -f
+```
+
+the CLI to check on the pods, you can use:
+
+```bash
+kubectl get pods -n demo
+```
+
+Hopefully everything has a status of `running`. If not, something is wrong.
+
+```bash
+NAME                                  READY   STATUS    RESTARTS   AGE
+blog-6896f6dd58-lzp5n                 1/1     Running   0          15m
+blog-neo4j-0                          1/1     Running   0          15m
+gateway-7f6d57765f-srb5d              1/1     Running   0          15m
+gateway-postgresql-74dbc8c987-zf9n7   1/1     Running   0          15m
+jhipster-registry-0                   1/1     Running   0          15m
+jhipster-registry-1                   1/1     Running   0          15m
+store-7bf788586b-6jq2f                1/1     Running   0          15m
+store-mongodb-0                       1/1     Running   0          15m
+```
+
+In K8s Lens, this looks like this:
+
+{% img blog/kubernetes-jhipster-aws/lens-status-green.png alt:"Lens status green" width:"1000" %}{: .center-image }
+
+
+Another useful command is `describe` (I won’t replicate the output here). This is great for more detailed information for debugging.
+
+```
+kubectl describe pods -n demo
```

Comment:
IMO describe is much better suited to be used with individual resources as there will be less noise and easier to find things `kubectl describe <pod|service|...> <resourse-name> -n demo`